api/python/provisioner/commands/upgrade/check_iso_authenticity.py

```diff
@@ -119,15 +119,23 @@ def run(self, iso_path: str, sig_file: str, gpg_pub_key: str = None,
         -------
 
         """
-        if import_pub_key and gpg_pub_key is not None:
-            self._import_gpg_public_key(gpg_pub_key)
-            gpg_pub_key = None  # NOTE: GPG public key is already imported
+        # to make converters and validators work
+        run_args = CheckISOAuthenticityArgs(
```

Comment:
Why do we need this parameters aggregation into the `CheckISOAuthenticityArgs` instance is we have this mechanism at the top of the class definition:
```python
_run_args_type = CheckISOAuthenticityArgs
```
So, maybe, it is better to remove this and leave only one approach?

Reply:
_run_args_types is used as a declaration of vars but is not called by generic logic so no converters and validators are applied, e.g. relative paths leaves as is at it failed in some cases during testing (gpg could not find a file)

Reply:
Ok, got it, thank you for your exploration. `_run_args_types` misled me, I thought that the args class is enough not only for parameters declaration and for parameters conversion and validation if setup `convertor` and validator `parameters` for the particular attribute.
Maybe, it is better to refer to `CheckISOAuthenticityArgs` via `self._run_args_type` as a suggestion.

Reply:
yes, agree. Generally the common logic should call that and pass the result to `run` methods instead of the current set of arguments.